### PR TITLE
Try various codecs instead of aborting reading a file

### PR DESCRIPTION
**Pull request by Cybolic**
Fix for #289

### CODE REVIEW

**Comment by Monnoroch** [comment]
This is not a good way to do it.
Look here: https://www.sublimetext.com/docs/3/api_reference.html.
You can actually get the current encoding from ST3!

**Comment by Cybolic** [comment]
I tried that, but I get Undefined from view.encoding(). I assume this is because we're reading the variables file, not the actual file in view, though I could be wrong.

**Comment by Monnoroch** [comment]
Oh, yeah. My bad.
Anyway, I think we probably should split the common case of utf8 and an exceptional case of everything else like this:
```
try:
    load in utf8 and return
except:
    try:
        load all others and return
    except:
        pass
return ""
```

Probably would be better to make a separate function for iterating through legacy encodings.

**Comment by Cybolic** [comment]
Apart from maybe saving a few bytes in memory for not initialising the list, I don't see much difference, but sure :)

**Comment by Monnoroch** [comment]
Well, I want to highlight the fact that this is not a normal execution path, but rather a best-effort fallback that we have to do for legacy users.

**Comment by Cybolic** [comment]
How's that?

**Comment by Monnoroch** [comment]
By moving it to a separate function with appropriate name.

File: ColorHighlighter.py [hunk 1]
```diff
@@ -137,10 +137,40 @@ def region_name(s, is_text):
 
 def read_file(fl):
     try:
-        with codecs.open(fl, "r", "utf-8") as f:
+        with codecs.open(fl, "r", "utf_8") as f:
```
Comment (Monnoroch) [inline ColorHighlighter.py]: Why did you change the encoding name?

File: ColorHighlighter.py [hunk 2]
```diff
@@ -137,10 +137,40 @@ def region_name(s, is_text):
...
             return f.read()
-    except UnicodeDecodeError as err:
-        print_error("read file error in file '"+fl+"':\n" + str(err))
+    except UnicodeDecodeError:
+        # Codecs supported by Python's codec module, in order of use according
+        # to http://w3techs.com/technologies/overview/character_encoding/all
+        encodings = [
```
Comment (Monnoroch) [inline ColorHighlighter.py]: As I said, it's probably would be better to move it to a separate function.

File: ColorHighlighter.py (hunk 1 above)
Comment (Cybolic) [inline ColorHighlighter.py]: I picked 'utf_8' as that's how it's listed in the Python documentation (https://docs.python.org/3/library/codecs.html), no other reason. I agree that utf-8 looks nice, so I'll switch it back :)

File: ColorHighlighter.py (hunk 2 above)
Comment (Cybolic) [inline ColorHighlighter.py]: All-right, will do.

Comment (Monnoroch) [inline ColorHighlighter.py]: Thanks. It's just that the plugin's code isn't all that great already, and I hope not to push it too far :)

**Comment by Monnoroch** [comment]
It seems we didn't fix the problem completely. #299
